puresasl/mechanisms.py

```diff
@@ -562,7 +562,7 @@ def wrap(self, outgoing):
                 protect = 1
             else:
                 protect = 0
-            kerberos.authGSSClientWrap(self.context, outgoing, None, protect)
+            kerberos.authGSSClientWrap(self.context, outgoing.decode('utf8'), None, protect)
```

Comment:
In Python 2.7, `outgoing.decode('utf8')` will create a unicode object when we want a str. So, I think this behavior needs to be conditional on the `PY3` variable defined in this module.

Reply:
No matter in Py2 or Py3, `outgoing.decode('utf8')` will create a unicode object and unicode object is acceptable by `authGSSClientWrap`. I've tested that in both Py2 and Py3.

Reply:
@thobbs